### DISCUSSION TITLE
Middleware works on local but not on Vercel

### DISCUSSION THREAD

**Discussion by chhuang** [vercel/community#909] [Help]
I'm using middleware to do a redirect. The redirect works on my local but it seems to be skipped/ignored on Vercel. What am I doing wrong? I'm using 12.3.1

```js
// middleware.ts
import { NextResponse } from 'next/server'
import type { NextRequest } from 'next/server'

const middleware = async (request: NextRequest) => {
  const response = NextResponse.next()

  const token = request.cookies.get('token')

  if (!token) {
    return NextResponse.redirect(new URL('/login', request.url))
  } else {
    return response
  }
}

export default middleware

export const config = {
  matcher: ['/', '/quests/:path*'],
}
```

![image](https://user-images.githubusercontent.com/78710/192916120-441cc708-4432-4a6a-9fc4-664d90e7b565.png)

![image](https://user-images.githubusercontent.com/78710/192935684-239d33d9-5194-4aca-bcb1-cfe2bb631cba.png)

**Comment by amyegan** [comment]
Thanks for sharing your middleware and output information @chhuang! I tried adding your middleware to a test project and I'm not able to reproduce the problem. Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can see what is going wrong?
